File: python/튜플(2019 kakao).py
def solution(s):
    answer = []
    arr = []
    # 튜플을 } 로 잘라서 탐색 시작
    for i in s.split("},"):
        # {, } 를 없애고 , 기준으로 push
        arr.append(i.replace("{", "").replace("}","").split(","))
    # 길이 순으로 정렬
    arr.sort(key=len)
    
    for i in arr:
        for j in i:
            if int(j) not in answer:
                answer.append(int(j))
    return answer

File: js/튜플(2019 kakao).js
// JSON.parse : 문자열 -> JS 객체로 변환
// let s = "{{2},{2,1},{2,1,3},{2,1,3,4}}";
// let newArr = s.replace(/{/g, '[').replace(/}/g, ']');
// console.log(newArr); // "[[2],[2,1],[2,1,3],[2,1,3,4]]" (문자열)

// console.log(newArr[0]); ->  "[" (문자열의 첫 번째 문자)

// let parsedArr = JSON.parse(newArr);
// console.log(parsedArr); -> [[2],[2,1],[2,1,3],[2,1,3,4]] (JavaScript 배열)

function solution(s) {
    var answer = [];
    let newArr = JSON.parse(s.replace(/{/g, '[').replace(/}/g, ']'));
    newArr.sort((a,b) => a.length - b.length);
    
    for (i of newArr) {
        for (j of i) {
            if (!answer.includes(j)) {
                answer.push(j);
            }
        }
    }
    return answer;
}